pkg/server/server.go

```diff
@@ -181,7 +224,7 @@ func NewServer(config *Config) (*KiamServer, error) {
 		arnResolver,
 	)
 	server.credentialsProvider = credentialsCache
-	server.manager = prefetch.NewManager(credentialsCache, server.pods, server.pods)
+	server.manager = prefetch.NewManager(credentialsCache, server.pods)
```

Comment:
One more potential big change to consider is moving the credential cache into the pod cache. Logically if you want unique credentials per pod, then maintaining two caches is less useful. If this is potentially a major version change to implement and it's high enough priority, should we consider grouping the change here?

Reply:
I'd defer picking it up until we do #38. I don't think the change would need a major version change to implement. I'd like us to sort #38 before we release v3 though- so we can still review the design for different pod and credential caches, the way sessions are named etc.